sonic_platform_base/sonic_xcvr/api/public/c_cmis.py

```diff
@@ -0,0 +1,901 @@
+"""
+    c-cmis.py
+
+    Implementation of XcvrApi that corresponds to C-CMIS
+"""
+from ...fields import consts
+from .cmis import CmisApi
+import time
+BYTELENGTH = 8
+
+class CCmisApi(CmisApi):
+    def __init__(self, xcvr_eeprom):
+        super(CCmisApi, self).__init__(xcvr_eeprom)
+
+    def get_freq_grid(self):
+        '''
+        This function returns the configured frequency grid. Unit in GHz
+        '''
+        freq_grid = self.xcvr_eeprom.read(consts.GRID_SPACING)
+        if freq_grid == 7:
+            return 75
+        elif freq_grid == 6:
+            return 33
+        elif freq_grid == 5:
+            return 100
+        elif freq_grid == 4:
+            return 50
+        elif freq_grid == 3:
+            return 25
+        elif freq_grid == 2:
+            return 12.5
+        elif freq_grid == 1:
+            return 6.25
+        elif freq_grid == 0:
+            return 3.125
+        else:
+            return None
+
+    def get_laser_config_freq(self):
+        '''
+        This function returns the configured laser frequency. Unit in GHz
+        '''
+        freq_grid = self.get_freq_grid()
+        channel = self.xcvr_eeprom.read(consts.LASER_CONFIG_CHANNEL)
+        if freq_grid == 75:
+            config_freq = 193100 + channel * freq_grid/3
+        else:
+            config_freq = 193100 + channel * freq_grid
+        return config_freq
+
+    def get_current_laser_freq(self):
+        '''
+        This function returns the monitored laser frequency. Unit in GHz
+        '''
+        return self.xcvr_eeprom.read(consts.LASER_CURRENT_FREQ)
+
+    def get_tuning_in_progress(self):
+        '''
+        This function returns tunning in progress status on media lane
+        0 means tuning not in progress
+        1 means tuning in progress
+        '''
+        return bool(self.xcvr_eeprom.read(consts.TUNING_IN_PROGRESS))
+
+    def get_wavelength_unlocked(self):
+        '''
+        This function returns wavelength unlocked status on media lane
+        0 means wavelength locked
+        1 means wavelength unlocked
+        '''
+        return bool(self.xcvr_eeprom.read(consts.WAVELENGTH_UNLOCKED))
```

Comment:
This is returning a bool instead of 0/1

Reply:
Changed the description to 

        '''
        This function returns wavelength unlocked status on media lane
        False means wavelength locked
        True means wavelength unlocked
        '''